src/games/strategy/triplea/delegate/BattleDelegateOrdered.java

```diff
@@ -0,0 +1,94 @@
+package games.strategy.triplea.delegate;
+
+import java.util.Collection;
+
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.delegate.BattleDelegate;
+import games.strategy.triplea.delegate.IBattle;
+import games.strategy.triplea.delegate.IBattle.BattleType;
+import games.strategy.util.Match;
+
+public class BattleDelegateOrdered extends BattleDelegate {
+
+  /**
+   * Called before the delegate will run.
+   */
+  @Override
+  public void start() {
+    super.start();
+    // Fight all air and bombing raids
+    for( final Territory t : m_battleTracker.getPendingBattleSites(true) ) {   // loop through bombing/air raids
+      final IBattle raid = m_battleTracker.getPendingBattle(t, true, null);             // Get air/bombing raid for current territory
+      if( raid == null ) {
+        try {
+          throw new Exception("Air/Bombing Raid gone missing in BattleDelegate");
+        } catch( Exception e ) {  // Will crash below
+        }
+      }
+      raid.fight(m_bridge);
+      final IBattle bombingRaid = m_battleTracker.getPendingBattle(t, true, BattleType.BOMBING_RAID );  // check to see if there's still a bombing raid for the territory - i.e. previous battle was an air raid
+      if( bombingRaid != null ) {
+        bombingRaid.fight(m_bridge);
+      }
+    }
+
+    // Fight all amphibious assaults with no retreat option for the attacker and no sea combat
+    int otherBattleCount = 0;
+    int amphibCount = 0;
+    IBattle lastAmphib = null;
```

Comment:
Still SRP violation with this variable. Variables should represent one logical concept. What this variable represents should not change depending upon where are in the code, it should mean exactly one thing.
Easiest way to fix is to not re-use the `lastAmphb` declaration, and use additional ones as needed.